Add DutyOfCare Procedure

mikejritter opened the pull request:
**What does this do?**
* Add dutyofcare recordtype

**Why are we doing this? (with JIRA link)**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1332

This is one of the procedures for the NAGPRA support we're working on for the next release.

**How should this be tested? Do these changes have associated tests?**
* Sanity check the changes first
  * `npm run lint`
  * `npm run test`
* Run the devserver
* Create a dutyofcare and check that all fields save
* Check the advanced search fields to make sure the messages are consistent

**Dependencies for merging? Releasing to production?**
I didn't any fields to the advanced search config. I was going to go through the fields and look for good candidates, but I know there's also been talk about reworking that page. Will bring it up in the weekly meeting as it seems like something good to discuss as a group.

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter tested against a local install

mikejritter commented on src/plugins/recordTypes/dutyofcare/messages.js:
```diff
@@ -0,0 +1,22 @@
+    name: {
+      id: 'record.dutyofcare.name',
+      description: 'The name of the record type',
+      defaultMessage: 'Duty of care',
```
I was wondering about casing for record names with multiple words. I'll get these updated.

mikejritter commented on src/plugins/recordTypes/dutyofcare/columns.js:
```diff
@@ -0,0 +1,17 @@
+      order: 10,
+      sortBy: 'dutyofcares_common:referenceNumber',
+      width: 200,
+    },
```
Looks like this is also missing `updatedAt` and the `narrow` definition. Going to get them in as well.

mikejritter replied:
Although looking at the other procedures I see they don't include `narrow` (although I thought some did). It doesn't seem like it will ever be used since it's only the related record sidebar that uses the `narrow` column type.

ray-lee replied:
You only need to define `narrow` if it's different than `default`. The sidebar will fall back if `narrow` doesn't exist. It's useful if you want to have fewer columns, or shorten the column labels to fit.

mikejritter commented on src/plugins/recordTypes/dutyofcare/fields.js:
```diff
@@ -0,0 +1,337 @@
+              messages: defineMessages({
+                name: {
+                  id: 'field.dutyofcares_common.dutyOfCareDetailsGroup.name',
+                  defaultMessage: 'Duty of care details',
```
I _think_ I may have been anticipating other nagpra procedures to have `detailGroupList`, but looking back at the specs I don't see any others. It seems like there are other fields that can be trimmed down a little bit as well (e.g. title), which I'll take care of.

mikejritter commented on src/plugins/recordTypes/dutyofcare/fields.js:
```diff
@@ -0,0 +1,337 @@
+              },
+              name: {
+                id: 'field.dutyofcares_common.dutyOfCareNumber.name',
+                defaultMessage: 'Duty of care id',
```
@ray-lee I updated the casing for this but just unresolved this - we don't really use ID that much in the forms and normally use `Identification/Reference/etc number` instead. I know for some of the others I had used that instead of `ID`, and I'm thinking of updating this as well to be `Duty of care number` to align better with other records.

mikejritter commented:
@ray-lee I just push a few commits that I believe resolves everything. I renamed the `ID` message to `number` and removed the `narrow` column definition to be more in line with the other procedures. 

On the app layer all the field names were simplified and missing attributes are present. In the services layer the xsd was updated for the field names.

ray-lee commented on src/plugins/recordTypes/dutyofcare/fields.js:
```diff
@@ -87,15 +87,11 @@ export default (configContext) => {
           },
         },
-        dutyOfCareTitle: {
+        title: {
```
My comment on the redundancy of "Duty of care title" was really just about the label, since people see it and I want to keep the UI from being too busy. I don't care all that much about the field name, so as far as I'm concerned you didn't need to change it. But it's fine that you did. Kristina may care more about how the fields are named.

mikejritter replied:
Yea, I've definitely been a little bit more paranoid about the field names from the qa testing. I didn't really see the need for `dutyofcare_commons.dutyofcare*` in the db when revising the fields, but I'll talk w/ Kristina about it as well for the upcoming procedures. 

I was planning on asking her for the data qa testing on dev once the procedures go up as well since it might be overwhelming to do the 6 or 7 new procedures we're getting at once.
